intake/source/tests/test_csv.py

```diff
@@ -98,6 +98,10 @@ def test_read_chunked(sample1_datasource, data_filenames):
 
 
 def check_read_pattern_output(source):
+    da = source.to_dask()
+    assert da.num.cat.known is True
```

Comment:
Presumably we can check the specific values of `da.num.cat` (the metadata), and also grab a partition and (`.get_partition(0).compute()`) and check that it also has the right categories.

Reply:
yeah we can. We check all that below though right?

Reply:
No, below we effectively do `.compute()` and then check everything on the resulting dataframe. I'm suggesting we should also check the `df` metadata before compute, and the contents of one of the partitions. It should be the same, of course.

Reply:
done